Add a recipe for the flycheck-projectile package

nbfalcon opened the pull request:
### Brief summary of what the package does

It implements a command to list all of a project's flycheck errors. The list automatically updates itself as new files are opened or when flycheck checks run. 

### Direct link to the package repository

https://github.com/nbfalcon/flycheck-projectile

### Your association with the package

I am the package's author and maintainer.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented:
Looks cool -- your recipe will need to exclude Elsafile.el.  Beyond that this is minor:

checkdoc (using version 0.6.1):
```
Warning (emacs): 
flycheck-projectile.el:90: Disambiguate flycheck-mode by preceding w/ function,command,variable,option or symbol.
Warning (emacs): 
flycheck-projectile.el:92: Lisp symbol `flycheck-mode' should appear in quotes
```

nbfalcon commented:
I added a :files block, which should solve the Elsafile issue. As for the checkdoc errors, I don't think there is anything I can do? Those ambiguous symbols refer to modes which always have variables, and I don't know of a way to ellegantly express that in a way that silences the checkdoc warnings. From looking at projectile, it has the same issue: mode symbols yield checkdoc errors.

As for the unqoted symbol at flycheck-projectile.el:92, I addressed that.

riscy commented:
Would a slightly different recipe work from your end?:
```elisp
(flycheck-projectile :repo "nbfalcon/flycheck-projectile" :fetcher github :files (:defaults (:exclude "Elsafile.el")))
```
It will give your package room to grow to include additional files if you decide to do so in the future, without opening another PR.  I think this is a quick merge once you decide.

nbfalcon commented:
It is a possibility. However, I don't plan to grow this package into more files; it is more likely that I will add additional Elsafile-like files in the future.
